src/server/bnetserver/REST/LoginRESTService.cpp

```diff
@@ -247,46 +247,104 @@ int32 LoginRESTService::HandlePost(soap* soapClient)
+                Field* fields = lastPlayerCharactersResult->Fetch();
+                Battlenet::RealmHandle realmId{ fields[1].GetUInt8(), fields[2].GetUInt8(), fields[3].GetUInt32() };
+                Battlenet::Session::LastPlayedCharacterInfo& lastPlayedCharacter = accountInfo->GameAccounts[fields[0].GetUInt32()]
+                    .LastPlayedCharacters[realmId.GetSubRegionAddress()];
 
-            } while (characterCountsResult->NextRow());
-        }
+                lastPlayedCharacter.RealmId = realmId;
+                lastPlayedCharacter.CharacterName = fields[4].GetString();
+                lastPlayedCharacter.CharacterGUID = fields[5].GetUInt64();
+                lastPlayedCharacter.LastPlayedTime = fields[6].GetUInt32();
+            }
 
-        stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_BNET_LAST_PLAYER_CHARACTERS);
-        stmt->setUInt32(0, accountInfo->Id);
-        if (PreparedQueryResult lastPlayerCharactersResult = LoginDatabase.Query(stmt))
-        {
-            Field* fields = lastPlayerCharactersResult->Fetch();
-            Battlenet::RealmHandle realmId{ fields[1].GetUInt8(), fields[2].GetUInt8(), fields[3].GetUInt32() };
-            Battlenet::Session::LastPlayedCharacterInfo& lastPlayedCharacter = accountInfo->GameAccounts[fields[0].GetUInt32()]
-                .LastPlayedCharacters[realmId.GetSubRegionAddress()];
-
-            lastPlayedCharacter.RealmId = realmId;
-            lastPlayedCharacter.CharacterName = fields[4].GetString();
-            lastPlayedCharacter.CharacterGUID = fields[5].GetUInt64();
-            lastPlayedCharacter.LastPlayedTime = fields[6].GetUInt32();
-        }
+            BigNumber ticket;
+            ticket.SetRand(20 * 8);
 
-        BigNumber ticket;
-        ticket.SetRand(20 * 8);
+            loginResult.set_login_ticket("TC-" + ByteArrayToHexStr(ticket.AsByteArray(20).get(), 20));
 
-        loginResult.set_login_ticket("TC-" + ByteArrayToHexStr(ticket.AsByteArray(20).get(), 20));
+            AddLoginTicket(loginResult.login_ticket(), std::move(accountInfo));
+        }
+        else if (!accountInfo->IsBanned)
+        {
+            uint32 const maxWrongPassword = uint32(sConfigMgr->GetIntDefault("WrongPass.MaxCount", 0));
 
-        AddLoginTicket(loginResult.login_ticket(), std::move(accountInfo));
+            if (sConfigMgr->GetBoolDefault("WrongPass.Logging", false))
+                TC_LOG_DEBUG("server.rest", "[%s, Account %s, Id %u] Attempted to connect with wrong password!", ip_address.c_str(), login.c_str(), accountInfo->Id);
+
+            if (maxWrongPassword)
+            {
+                SQLTransaction trans = LoginDatabase.BeginTransaction();
+                
+                PreparedStatement* updateFailedLogins;
+                PreparedStatement* insertBan = nullptr;
+                PreparedStatement* resetFailedLogins = nullptr;
```

Comment:
When I watched `SQLTransaction::Append()` It looked like it makes a pointer copy that's why I have 3 pointers.